app/src/main/java/org/mozilla/fenix/library/recentlyclosed/RecentlyClosedAdapter.kt

```diff
@@ -31,6 +36,38 @@ class RecentlyClosedAdapter(
             oldItem.id == newItem.id
 
         override fun areContentsTheSame(oldItem: RecoverableTab, newItem: RecoverableTab) =
-            oldItem.id == newItem.id
+            oldItem == newItem
+    }
+
+    fun updateData(tabs: List<RecoverableTab>, selectedTabs: Set<RecoverableTab>) {
```

Comment:
nit: Let's organize the methods so that private bindings are together, overridden methods are together, and private methods are together.